app/src/userpages/modules/userPageStreams/services.js

```diff
@@ -20,3 +30,38 @@ export const getMyStreamPermissions = (id: StreamId): ApiResult<Array<Permission
     get(formatApiUrl('streams', id, 'permissions', 'me'))
 
 export const getStreamStatus = (id: StreamId): ApiResult<StreamStatus> => get(formatApiUrl('streams', id, 'status'))
+
+export const uploadCsvFile = (id: StreamId, file: File): Promise<CsvUploadResult> => {
+    const options = {
+        headers: {
+            'Content-Type': 'multipart/form-data',
+        },
+    }
+
+    const formData = new FormData()
+    formData.append('file', file)
+
+    return post(formatApiUrl('streams', id, 'uploadCsvFile'), formData, options)
+}
+
+export const confirmCsvFileUpload = (
+    id: StreamId,
+    fileId: string,
+    dateFormat: string,
+    timestampColumnIndex: number,
+): ApiResult<any> => (
+    post(formatApiUrl('streams', id, 'confirmCsvFileUpload'), {
+        fileId,
+        dateFormat,
+        timestampColumnIndex,
+    })
+)
+
+export const getRange = (id: StreamId): ApiResult<Range> => get(formatApiUrl('streams', id, 'range'))
+
+export const deleteDataUpTo = (id: StreamId, date: Date): ApiResult<any> => (
+    del(formatApiUrl('stream', id, 'deleteDataUpTo'), {
+        id,
+        date: moment(date).format('YYYY-MM-DD'),
```

Comment:
New format for the backend dates (for stream delete data endpoints) is:
2019-05-10T11:45:40Z (old yyyy-mm-dd)
or
1557488802759 -> milliseconds from epoch (new feature)